__tests__/index.test.js

```diff
@@ -78,6 +78,52 @@ describe('/api/articles', () => {
       }
     })
   })
+
+  describe('GET:200 allows articles to be filtered by topic', () => {
+    it('returning matching topics for a present search term', async () => {
+      const res = await request(app).get('/api/articles?topic=mitch').expect(200)
+      const { articles } = res.body
+
+      const filteredArticles = testArticles
+        .filter((article) => article.topic === 'mitch')
+        .sort((a1, a2) => a2.created_at - a1.created_at)
+        .map((article) => {
+          const articleCopy = { ...article }
+          delete articleCopy.body
+          delete articleCopy.created_at
+
+          return articleCopy
+        })
+
+      articles.forEach((article, i) => {
+        expect(article).toMatchObject(filteredArticles[i])
+      })
+    })
+
+    it('returning no topics for an absent search term', async () => {
+      const res = await request(app).get('/api/articles?topic=NoSuchTopicBro').expect(200)
+      const { articles } = res.body
+
+      expect(articles).toHaveLength(0)
+    })
+
+    it('returning all articles for an invalid query parameter', async () => {
+      const res = await request(app).get('/api/articles?topi=banana').expect(200)
+      const { articles } = res.body
+
```

Comment:
Shouldn't this be throwing an error not getting 200?

Reply:
Fixed